Make fail2ban work again

rvl opened the pull request:
1. fail2ban depends on the systemd python module which has changed location.
2. There was some problem with the location of installed files (issue #11534).
3. Update to the latest release (0.9.3 2015/08/01 - lets-all-stay-friends) for good measure.

rvl committed:
fail2ban: 0.9.1 -> 0.9.3

rvl committed:
python systemd: init at 231

This module has been split out from the main systemd repository.

rvl committed:
systemd: python module split out since v223

rvl committed:
fail2ban: python packaging workaround for /etc/fail2ban installation (fixes #11534)

Removed path substitutions from setup.py because these should be handled
by the setuptools install prefix.

Except that the install prefix won't quite work until issue #4968 is
resolved.

In the meantime there are preInstall and postInstall scripts so that
this package continues to work with the nix python packaging
improvements.

rvl committed:
fail2ban: update python-systemd dependency

The python module has been split off from systemd since v223.

cillianderoiste commented:
@rvl the Travis build fails with: 

    error: [Errno 2] No such file or directory: '/usr/include/systemd/sd-messages.h'

Do you have chroot building enabled for nix? That might help to reproduce it.

FRidh commented on pkgs/tools/security/fail2ban/default.nix:
```diff
@@ -9,22 +9,17 @@ pythonPackages.buildPythonPackage {
   buildInputs = [ unzip ];
 
-  pythonPath = (stdenv.lib.optional stdenv.isLinux systemd)
+  pythonPath = (stdenv.lib.optional stdenv.isLinux pythonPackages.systemd)
```
could you use `propagatedBuildInputs` here

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -18314,21 +18313,53 @@ in modules // {
+  systemd = buildPythonPackage rec {
+    name = "python-systemd-231";
+
+    src = pkgs.fetchgit {
```
how about using a release with fetchurl?
https://github.com/systemd/python-systemd/releases/tag/v231

rvl commented:
@cillianderoiste Thanks for the tip. I tried chroot building just then but couldn't get it to work. As a workaround I have removed `libsystemd-dev` from the debian system which I am building from. My other problem was that the travis build uses `cc` but my own builds use `gcc`.

@FRidh I have implemented your suggestions, thanks.

jeaye commented:
Looking forward to getting fail2ban working again (China really wants my root account, it seems); thanks for this.

rvl commented:
No prob @jeaye.

Can I ask @domenkozar to kindly check the `preConfigure`/`postInstall` change (b61e057) for us?

domenkozar commented:
@rvl looks like https://github.com/NixOS/nixpkgs/issues/4968 that provides the current workaround method

rvl commented:
Ok great @domenkozar that explains it. I have changed it to use that workaround. However this `setup.py` was not as easy to sed, so there is a `postInstall` cleanup.

garbas commented:
@rvl tnx!

rvl commented:
@garbas thx for merging
